hoss/spatial.py

```diff
@@ -187,6 +216,73 @@ def get_projected_x_y_index_ranges(
     return x_y_index_ranges
 
 
+def get_x_y_index_ranges_from_coordinates(
+    non_spatial_variable: str,
+    varinfo: VarInfoFromDmr,
+    prefetch_coordinate_datasets: Dataset,
+    latitude_coordinate: VariableFromDmr,
+    longitude_coordinate: VariableFromDmr,
+    index_ranges: IndexRanges,
+    bounding_box: BBox = None,
+    shape_file_path: str = None,
+) -> IndexRanges:
+    """This function returns a dictionary containing the minimum and maximum
+    index ranges for the projected_x and projected_y recalculated dimension scales
+
+    index_ranges = {'projected_x': (20, 42), 'projected_y': (31, 53)}
+
+    This method is called when the CF standards are not followed
+    in the source granule and only coordinate datasets are provided.
+    The coordinate datasets along with the crs is used to calculate
+    the x-y projected dimension scales. The dimensions of the input,
+    non-spatial variable are checked for associated coordinates. If
+    these are present, and they have not already been added to the
+    `index_ranges` cache, the extents of the input spatial subset
+    are determined in these projected coordinates. The minimum and
+    maximum values are then derived from these projected coordinate
+    points.
+
+    """
+
+    crs = get_variable_crs(non_spatial_variable, varinfo)
+    projected_x = 'projected_x'
+    projected_y = 'projected_y'
+    override_dimension_datasets = update_dimension_variables(
```

Comment:
Please can you be consistent with the rest of the code and use "variable" to refer to variables, and reserve "dataset" to refer to the `netCDF4.Dataset` object. Using "datasets" to mean variables will cause confusion in the future. (I acknowledge that "dataset" is also often used to mean "variable" by other software, but it is not so far used that way in the HOSS code, due to the terminology of the `netCDF4` Python package)

Reply:
done - 0666248f0d96e91bf8a900c598be18b674b1cfa9

Reply:
Thanks for the update. I took a quick skim and there's one place remaining that uses `dataset` instead of variable. It's the `get_valid_indices_in_dataset` function. Otherwise, I think the language is now consistent.

Reply:
the update_dimension_variables is now called create_dimension_scales_from_coordinates based on David's feedback (will be in the new commits in the new branch)

the get_valid_indices_in_dataset does get the valid indices from the numpy array passed in. Not sure if we can call it get_valid_indices_in_variable...